File: ddd/tests/fixtures/operation-error-set/typescript-class-workspace/billing-domain/src/unreferenced.ts
/** Both error types are imported from a module the snapshot does not contain. */

import type { IssueInvoiceError, OpenInvoiceError } from "./absent.ts";
import type { Result } from "./result.ts";

export class Invoice {
  static open(): Result<Invoice, OpenInvoiceError> {
    return { ok: true, value: new Invoice() };
  }

  issue(): Result<void, IssueInvoiceError> {
    return { ok: true, value: undefined };
  }
}

File: ddd/tests/fixtures/operation-error-set/rust-workspace/billing-domain/src/unreferenced.rs
//! Both error types are imported from a module the snapshot does not contain.

use crate::absent::IssueInvoiceError;
use crate::absent::OpenInvoiceError;

pub struct Invoice;

impl Invoice {
    pub fn issue(&mut self) -> Result<(), IssueInvoiceError> {
        Ok(())
    }

    pub fn open() -> Result<Self, OpenInvoiceError> {
        Ok(Self)
    }
}
